app/components/animated_number/index.test.tsx

```diff
@@ -0,0 +1,176 @@
+            const {getByTestId} = render(<AnimatedNumber animateToNumber={animateToNumber}/>);
+
+            const text = getByTestId('no-animation-number');
+
+            fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+            const checkEachDigit = animateToNumber.toString().split('').map(async (number, index) => {
+                const useIndex = animateToNumber.toString().length - 1 - index;
+
+                let translateY = -1;
+
+                // every digit will have a row of 10 numbers, so the translateY should be the height of the number * the number * -1 (since the animation is going up)
+                await waitFor(() => {
+                    const transformedView = getByTestId(`animated-number-view-${useIndex}`);
+
+                    translateY = transformedView.props.style.transform[0].translateY;
+                });
+
+                expect(Math.abs(translateY / NUMBER_HEIGHT)).toEqual(Number(number));
+            });
+
+            await Promise.all(checkEachDigit);
+        });
+    });
+
+    describe.each([146, 144, 1, 1000000, -145])('should rerender the correct number that it animates to', (animateToNumber: number) => {
+        it(`should display the number ${animateToNumber}`, async () => {
+            const startingNumber = 145;
+            const {getByTestId, rerender} = render(<AnimatedNumber animateToNumber={startingNumber}/>);
+
+            const text = getByTestId('no-animation-number');
+
+            fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+            rerender(<AnimatedNumber animateToNumber={animateToNumber}/>);
+
+            const animateToNumberString = Math.abs(animateToNumber).toString();
+            const checkEachDigit = animateToNumberString.split('').map(async (number, index) => {
+                const useIndex = animateToNumberString.length - 1 - index;
+
+                const transformedView = getByTestId(`animated-number-view-${useIndex}`);
+                const translateY = transformedView.props.style.transform[0].translateY;
+
+                expect(Math.abs((translateY) / NUMBER_HEIGHT)).toEqual(Number(number));
+            });
+
+            await Promise.all(checkEachDigit);
+        });
+    });
+
+    it('KNOWN UI BUG: should show that there will be an issue if the text height changes, due to the non-animated number view has been removed', async () => {
+        // the number text will get cut-off if the user changes the text size on their mobile devices
+        const {getByTestId} = render(<AnimatedNumber animateToNumber={123}/>);
+
+        const text = getByTestId('no-animation-number');
+
+        fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+
+        try {
+            fireEvent(text, 'onLayout', {nativeEvent: {layout: {height: NUMBER_HEIGHT}}});
+        } catch (e) {
+            expect(e).toBeTruthy();
```

Comment:
Because fireEvent() is not a promise and there's no rejection.

It still throws an error, because the View is no longer there (since it has been removed).